assets/app/scripts/controllers/createRoute.js

```diff
@@ -0,0 +1,91 @@
+'use strict';
+
+/**
+ * @ngdoc function
+ * @name openshiftConsole.controller:CreateRouteController
+ * @description
+ * # CreateRouteController
+ * Controller of the openshiftConsole
+ */
+angular.module('openshiftConsole')
+  .controller('CreateRouteController', function ($filter, $routeParams, $scope, $window, ApplicationGenerator, DataService, Navigate, project) {
+    $scope.alerts = {};
+    $scope.renderOptions = $scope.renderOptions || {};
+    $scope.renderOptions.hideFilterWidget = true;
+    $scope.projectName = $routeParams.project;
+    $scope.serviceName = $routeParams.service;
+
+    // Prefill route name with the service name.
+    $scope.routing = {
+      name: $scope.serviceName || ""
+    };
+
+    project
+      .get($routeParams.project)
+      .then(_.spread(function(project, context) {
+        $scope.project = project;
+
+        var updatePortOptions = function(service) {
+          if (!service) {
+            return;
+          }
+
+          $scope.routing.portOptions = _.map(service.spec.ports, function(portMapping) {
+            return {
+              containerPort: portMapping.targetPort,
+              protocol: portMapping.protocol
+            };
+          });
+
+          if ($scope.routing.portOptions.length) {
+            $scope.routing.targetPort = $scope.routing.portOptions[0];
```

Comment:
should we be defaulting to the named 'http' port if its there like we do in create from image?